pkg/reconciler/v1alpha1/configuration/configuration_test.go

```diff
@@ -71,209 +65,129 @@ func TestReconcile(t *testing.T) {
 			cfg("no-revisions-yet", "foo", 1234),
 		},
 		WantCreates: []metav1.Object{
-			resources.MakeRevision(cfg("no-revisions-yet", "foo", 1234), nil),
+			rev("no-revisions-yet", "foo", 1234),
 		},
 		WantUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: cfgWithStatus("no-revisions-yet", "foo", 1234,
-				v1alpha1.ConfigurationStatus{
-					LatestCreatedRevisionName: "no-revisions-yet-01234",
-					ObservedGeneration:        1234,
-					Conditions: duckv1alpha1.Conditions{{
-						Type:     v1alpha1.ConfigurationConditionReady,
-						Status:   corev1.ConditionUnknown,
-						Severity: "Error",
-					}},
-				}),
+			Object: cfg("no-revisions-yet", "foo", 1234,
+				// The following properties are set when we first reconcile a
+				// Configuration and a Revision is created.
+				WithLatestCreated, WithObservedGen),
 		}},
 		Key: "foo/no-revisions-yet",
 	}, {
 		Name: "webhook validation failure",
 		// If we attempt to create a Revision with a bad ConcurrencyModel set, we fail.
 		WantErr: true,
 		Objects: []runtime.Object{
-			setConcurrencyModel(cfg("validation-failure", "foo", 1234), "Bogus"),
+			cfg("validation-failure", "foo", 1234, WithConfigConcurrencyModel("Bogus")),
 		},
 		WantCreates: []metav1.Object{
-			setRevConcurrencyModel(resources.MakeRevision(cfg("validation-failure", "foo", 1234), nil), "Bogus"),
+			rev("validation-failure", "foo", 1234, WithRevConcurrencyModel("Bogus")),
 		},
 		WantUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: setConcurrencyModel(cfgWithStatus("validation-failure", "foo", 1234,
-				v1alpha1.ConfigurationStatus{
-					Conditions: duckv1alpha1.Conditions{{
-						Type:     v1alpha1.ConfigurationConditionReady,
-						Status:   corev1.ConditionFalse,
-						Reason:   "RevisionFailed",
-						Message:  `Revision creation failed with message: "invalid value \"Bogus\": spec.concurrencyModel".`,
-						Severity: "Error",
-					}},
-				}), "Bogus"),
+			Object: cfg("validation-failure", "foo", 1234, WithConfigConcurrencyModel("Bogus"),
+				// Expect Revision creation to fail with the following error.
+				MarkRevisionCreationFailed(`invalid value "Bogus": spec.concurrencyModel`)),
```

Comment:
This looks to be using `cfg.Status.MarkRevisionCreationFailed(msg)`. I am not sure if this method is tested elsewhere, but the previous logic had some coverage of the method validating that the message is transformed and all fields are set.

Reply:
My thinking is that this is a gap in `configuration_types_test.go` and not a gap here.

Opened: https://github.com/knative/serving/issues/2499